bin/helpers/utils.js

```diff
@@ -586,6 +586,26 @@ exports.versionChangedMessage = (preferredVersion, actualVersion) => {
   return message
 }
 
+exports.invalidLocalMode = (bsConfig, args) => {
+  let localMode = undefined;
+  if (!this.isUndefined(args.localMode)) {
+    localMode = args.localMode;
+  } else if (!this.isUndefined(bsConfig.connection_settings)) {
+    localMode = bsConfig.connection_settings.local_mode
+  }
+  return (this.searchForOption('--local-mode') || !this.isUndefined(localMode)) && !["always-on","on-demand"].includes(localMode);
```

Comment:
The condition is wrong. Please look at previous validation : 
Utils.searchForOption('--local-mode') && ( Utils.isUndefined(args.localMode) || (!Utils.isUndefined(args.localMode) && !["always-on","on-demand"].includes(args.localMode)))